WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -217,13 +222,46 @@ public void onSiteRemoved(OnSiteRemoved event) {
     @SuppressWarnings("unused")
     @Subscribe(threadMode = ThreadMode.MAIN)
     public void onSiteChanged(OnSiteChanged event) {
-        mDebouncer.debounce(Void.class, new Runnable() {
-            @Override public void run() {
-                if (!isFinishing()) {
-                    getAdapter().loadSites();
-                }
+        if (mSwipeToRefreshHelper.isRefreshing()) {
+            mSwipeToRefreshHelper.setRefreshing(false);
+            if (!isFinishing()) {
+                getAdapter().loadSites();
             }
-        }, 200, TimeUnit.MILLISECONDS);
+        } else {
+            mDebouncer.debounce(Void.class, new Runnable() {
+                @Override
+                public void run() {
+                    if (!isFinishing()) {
+                        getAdapter().loadSites();
+                    }
+                }
+            }, 200, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private void initSwipeToRefreshHelper(View view) {
+        if (view == null) {
+            return;
+        }
+        mSwipeToRefreshHelper = new SwipeToRefreshHelper(
+                this,
+                (CustomSwipeRefreshLayout) view.findViewById(R.id.ptr_layout),
+                new SwipeToRefreshHelper.RefreshListener() {
+                    @Override
+                    public void onRefreshStarted() {
+                        if (isFinishing()) {
+                            return;
+                        }
+                        if (!NetworkUtils.checkConnection(SitePickerActivity.this)) {
+                            mSwipeToRefreshHelper.setRefreshing(false);
+                            ToastUtils.showToast(SitePickerActivity.this, R.string.site_picker_no_network_refresh_error);
+                            return;
+                        }
+                        mDispatcher.dispatch(SiteActionBuilder.newRemoveWpcomAndJetpackSitesAction());
```

Comment:
I agree that we probably shouldn't do a mass removal here - and I wanted to add that it would be better to handle clearing any remotely-removed sites in FluxC instead (this way we'll also have the same behavior here and for the existing 15 minute auto-refresh of the list).

So perhaps after a `FETCH_SITES` completes, we track WP.com sites linked to that account that weren't in the new fetch, and remove them after we finish updating the db (all in FluxC).